### DISCUSSION TITLE
Clearing the evaluations history

### DISCUSSION THREAD

**Discussion by n6s** [open-webui/open-webui#7938] [Show and tell]
For me, going to `Admin Panel > Evaluations` had stopped working. The page would never finish loading and show the Leaderboard as it used to and sometimes I'd get a page crash, like "Aw Snap! Error 5". 

I searched here and didn't see any Issues or Discussions about it. A long story short, here's how I cleared that evaluations database.
* Authenticate:
```
curl -X POST http://127.0.0.1:8080/api/v1/auths/signin -H "Content-Type: application/json" -d '{"email": "mysernamehere", "password": "mypasswordhere"}' -c cookies.txt
```
* List All Feedbacks:
```
curl -b cookies.txt http://127.0.0.1:8080/api/v1/evaluations/feedbacks/all
```
* Delete All Feedbacks:
```
curl -X DELETE http://127.0.0.1:8080/api/v1/evaluations/feedbacks/all -b cookies.txt
```

I would be interested in your easier and more elegant solutions.

**Comment by ColinMaximillion** [comment]
Your solution is still the only one I can find, and it works wonders for me. Thank you, n6s.

**Comment by EgShes** [comment]
I had the same issue — the Evaluations page wouldn't load due to too many old feedbacks.

The solution from the original post works, but you can delete individual feedbacks with singular /feedback/{id}:

DELETE /api/v1/evaluations/feedback/{id}

Here's a Python script to delete feedbacks filtered by date and model:

```python
import json
from datetime import datetime
import requests

API_KEY = "your-api-key"
BASE_URL = "https://your-openwebui-instance"
CUTOFF = datetime(2025, 11, 1)  # delete everything before this date

headers = {"Authorization": f"Bearer {API_KEY}"}

# Step 1: Export feedbacks
# GET /api/v1/evaluations/feedbacks/all > feedbacks.json

with open("feedbacks.json") as f:
    feedbacks = json.load(f)

for fb in feedbacks:
    ts = fb.get("updated_at") or fb.get("created_at", 0)
    dt = datetime.fromtimestamp(ts) if isinstance(ts, (int, float)) else datetime.fromisoformat(ts)

    if dt < CUTOFF:
        fb_id = fb["id"]
        r = requests.delete(
            f"{BASE_URL}/api/v1/evaluations/feedback/{fb_id}", headers=headers
        )
        print(f"Deleted {fb_id} from {dt}: {r.status_code}")
```

You can also add extra filters like fb["data"]["model_id"] == "some-model" to only delete feedbacks for a specific model.
